rules/repo.bzl

```diff
@@ -4,12 +4,17 @@
 
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-def http_archive_or_local(**kwargs):
-    local = kwargs.pop("local", None)
-    if local:
-        native.local_repository(
+def http_archive_or_local(local = None, build_file = None, **kwargs):
+    if not local:
+        http_archive(build_file = build_file, **kwargs)
+    elif build_file:
+        native.new_local_repository(
```

Comment:
I assume http_archive doesn't need or properly handle kwargs here so it's intentionally omitted.

Reply:
The diff looks a little strange, but `http_archive()` should still get identical parameters to before. The only intended behavior difference is that we're sometimes calling `new_local_repository()` instead of `local_repository()`.

Reply:
Ugh, typo, sorry

Reply:
Ah! I don't see any reason we couldn't pass kwargs to `new_local_repository` (and `local_repository` for that matter). I'd prefer to punt on this because it's a separate behavior change. WDYT?

Reply:
@drewmacrae I'll be OOO tomorrow. Would you please merge this PR if you don't feel strongly about kwargs?